### PR TITLE
initial check-in of basic skeleton/BL

### PR DESCRIPTION
**Pull request by craigmcnally**
See [MODORDERS-21](https://issues.folio.org/projects/MODORDERS/issues/MODORDERS-21) for details.

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/PostOrdersHelper.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+              futures.add(polFut);
+            } catch (Exception e) {
+              logger.error("Exception calling POST /po_line", e);
+              throw new CompletionException(e);
+            }
+          });
+
+          VertxCompletableFuture.allOf(ctx, futures.toArray(new CompletableFuture[futures.size()]))
+            .thenAccept(v -> {
+              compPO.setPoLines(lines);
```
Comment (mreno-EBSCO) [inline src/main/java/org/folio/rest/impl/PostOrdersHelper.java]: We should determine if the order of the returned PO lines, in multi line POs, matters, i.e. should match what is sent in the POST?

File: src/main/java/org/folio/rest/impl/PostOrdersHelper.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+          String poId = po.getId();
+          compPO.setPurchaseOrder(po);
+
+          List<PoLine> lines = new ArrayList<>(compPO.getPoLines().size());
+          List<CompletableFuture<Void>> futures = new ArrayList<>();
+          compPO.getPoLines().forEach(line -> {
+            line.setPurchaseOrderId(poId);
+            try {
+              Buffer polBuf = JsonObject.mapFrom(line).toBuffer();
+              CompletableFuture<Void> polFut = httpClient.request(HttpMethod.POST, polBuf, "/po_line", null)
```
Comment (mreno-EBSCO) [inline src/main/java/org/folio/rest/impl/PostOrdersHelper.java]: The Okapi headers need to be passed in here instead of `null`.

File: pom.xml [hunk 3]
```diff
@@ -2,7 +2,7 @@
   <modelVersion>4.0.0</modelVersion>
   <groupId>org.folio</groupId>
   <artifactId>mod-orders</artifactId>
-  <version>1.0.0-SNAPSHOT</version>
+  <version>1.0.1-SNAPSHOT</version>
```
Comment (mreno-EBSCO) [inline pom.xml]: NEWS.md should be updated to reflect 1.0.1 is the unreleased version.

File: src/main/java/org/folio/rest/impl/PostOrdersHelper.java (hunk 1 above)
Comment (craigmcnally) [reply src/main/java/org/folio/rest/impl/PostOrdersHelper.java]: I'll bring this up at the next meeting w/ Stacks

**Pull request merged**